tests/main/microk8s-smoke/task.yaml

```diff
@@ -40,6 +40,19 @@ prepare: |
     cp /etc/systemd/system/snapd.service.d/local.conf /etc/systemd/system/snapd.service.d/local.conf.bak
     sed 's/SNAPD_CONFIGURE_HOOK_TIMEOUT=.*s/SNAPD_CONFIGURE_HOOK_TIMEOUT=180s/g' -i /etc/systemd/system/snapd.service.d/local.conf
     tests.cleanup defer mv /etc/systemd/system/snapd.service.d/local.conf.bak /etc/systemd/system/snapd.service.d/local.conf
+
+    if [ -n "${http_proxy:-}" ] || [ -n "${https_proxy:-}" ] ; then
+        # Configuration done following https://microk8s.io/docs/install-proxy
+        cp /etc/environment /etc/environment.bak
+        echo "http_proxy=$http_proxy" >> /etc/environment
+        echo "HTTP_PROXY=$https_proxy" >> /etc/environment
+        echo "https_proxy=$https_proxy" >> /etc/environment
+        echo "HTTPS_PROXY=$https_proxy" >> /etc/environment
+        echo "NO_PROXY=10.0.0.0/8,192.168.0.0/16,127.0.0.1,172.16.0.0/16,.svc,localhost"  >> /etc/environment
+        echo "no_proxy=10.0.0.0/8,192.168.0.0/16,127.0.0.1,172.16.0.0/16,.svc,localhost"  >> /etc/environment
```

Comment:
It seems that `http_proxy`, `HTTP_PROXY`, `https_proxy`, `HTTPS_PROXY` are all properly set in the openstack backend https://github.com/canonical/snapd/blob/5a782b1dd633369387a53ab58722980a2520202f/tests/lib/spread/backend.openstack.yaml#L7-L12

Maybe we only need to update `no_proxy` and `NO_PROXY` to skip the IP ranges for pods and services with a comment explaining this?